runelite-client/src/main/java/net/runelite/client/plugins/inventorytagger/InventoryTaggerPlugin.java

```diff
@@ -0,0 +1,178 @@
+package net.runelite.client.plugins.inventorytagger;
+
+import com.google.common.eventbus.Subscribe;
+import java.awt.Color;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuOpened;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.util.Text;
+import org.apache.commons.lang3.ArrayUtils;
+
+@PluginDescriptor(
+	name = "Inventory Item Tagging",
+	enabledByDefault = true
+)
+public class InventoryTaggerPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private InventoryTaggerPlugin plugin;
+
+	@Inject
+	private InventoryTaggerOverlay overlay;
+
+	public Map<String, TaggedItems> taggedItems = new HashMap<>();
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	private boolean editorMode = false;
+
+	private static final String SETNAME_MELEE = "Melee";
+	private static final String SETNAME_RANGE = "Ranged";
+	private static final String SETNAME_MAGE = "Mage";
+	private static final String SETNAME_OTHER = "Other";
+
+	private static final String CONFIGURE = "Configure";
+	private static final String SAVE = "Save";
+	private static final String MENU_TARGET = "<col=ff9040>Inventory Tagger";
+
+	private static final String MENU_SET = "Set";
+	private static final String MENU_REMOVE = "Remove";
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		taggedItems = new HashMap<>();
+		taggedItems.put(SETNAME_MELEE, new TaggedItems(SETNAME_MELEE, new Color(255, 0, 0, 255)));
```

Comment:
I would just name these fields something like "Alpha", "Beta", "Gama", "Delta" so the names would be more generic